File: typescript/src/CashMachine.ts
import { BranchFinder } from './BranchFinder';

export class CashMachine {
  private readonly branchFinder = new BranchFinder();

  withdraw({ cashAmount, customerName, town }: {
    town: string;
    customerName: string;
    cashAmount: number;
  }): boolean {
    return this.branchFinder
      .findBranchForTown(town)
      .personalAccounts
      .getAccountForCustomer(customerName)
      .withdraw(cashAmount);
  }
}

File: c/src/cash_machine.c

#include "cash_machine.h"
#include "branch_finder.h"
#include "personal_accounts.h"
#include "account.h"

bool withdraw(int town_code, int customer_id, Cash* cashAmount)
{
    return withdraw_from_account(get_account_for_customer(personal_accounts_for_branch(find_branch_for_town(town_code)), customer_id), cashAmount);
}
